README.md

```diff
@@ -1,30 +1,77 @@
 # rav1d
 
-**rav1d** is an **AV1** cross-platform **d**ecoder, open-source, and focused on speed and correctness. It is a Rust port of [dav1d](https://code.videolan.org/videolan/dav1d).
+**rav1d** is an AV1 cross-platform decoder, open-source, and focused on speed and correctness. It is a Rust port of [dav1d](https://code.videolan.org/videolan/dav1d).
 
-rav1d is currently experimental. Core functionality has been transpiled using [c2rust](https://github.com/immunant/c2rust), but not everything has been ported yet and the transpiled code needs to be cleaned up from unsafe transpiled Rust to safe, idiomatic Rust.
+# Building
+
+rav1d is written in Rust and uses the standard Rust toolchain to build. The Rust toolchain can be installed by going to https://rustup.rs. rav1d currently builds with a nightly compiler, which is specified in `rust-toolchain.toml`. The correct nightly build will automatically be installed for you when you run `cargo build`.
+
+For x86 targets you'll also need to install [nasm](https://nasm.us/) in order to build with assembly support.
+
+A release build can then be made using cargo:
+
+```txt
+cargo build --release
+```
+
+For development purposes you may also want to use the opt-dev profile, which runs faster than a regular debug build but has all debug checks still enabled:
+
+```txt
+cargo build --profile opt-dev
+```
+
+## Cross-Compiling
+
+rav1d can be cross-compiled for a target other than the host platform using the Cargo `--target` flag. This may require passing additional arguments to the Rust compiler to tell it what linker to use. This can be done by setting the `RUSTFLAGS` enviroment variable and specifying the `linker` compiler flag. You'll also need to specify the `+crt-static` target feature. For example, compiling for `aarch64-unknown-linux-gnu` from a Ubuntu Linux machine would be done as follows:
+
+```txt
```

Comment:
```suggestion
```sh
```

Reply:
Same applies to all of the other `txt`s.  They can be `sh` if it's directly a shell command.  If it's shell input and output, with a `> ` or `$ ` and then the output, you can use `shell` as the language.